lib/pure/random.nim

```diff
@@ -171,34 +165,30 @@ proc skipRandomNumbers*(s: var Rand) =
   ## Rand state to a thread, call this proc before passing it to the next one.
   ## By using the Rand state this way, the subsequences of random numbers
   ## generated in each thread will never overlap as long as no thread generates
-  ## more than 2^64 random numbers.
-  ##
-  ## The following example below demonstrates this pattern:
-  ##
-  ## .. code-block::
-  ##   # Compile this example with --threads:on
-  ##   import random
-  ##   import threadpool
+  ## more than `2^64` random numbers.
   ##
-  ##   const spawns = 4
-  ##   const numbers = 100000
-  ##
-  ##   proc randomSum(rand: Rand): int =
-  ##     var r = rand
-  ##     for i in 1..numbers:
-  ##       result += rand(1..10)
-  ##
-  ##   var r = initRand(2019)
-  ##   var vals: array[spawns, FlowVar[int]]
-  ##   for val in vals.mitems:
-  ##     val = spawn(randomSum(r))
-  ##     r.skipRandomNumbers()
-  ##
-  ##   for val in vals:
-  ##     echo ^val
-  ##
-  ## See also:
+  ## **See also:**
   ## * `next proc<#next,Rand>`_
+  runnableExamples("--threads:on"):
+    import std/[random, threadpool]
+
+    const spawns = 4
+    const numbers = 100000
+
+    proc randomSum(rand: Rand): int =
+      var r = rand
+      for i in 1..numbers:
+        result += rand(0..10)
```

Comment:
pre-existing but I don't understand this code.

Reply:
Since I wrote this, yeah I just needed something that wouldn't take a whole page (the rest of the code is already long enough for an example) so I wrote something really simple and boring, which is "let's just add a bunch of random numbers together". It might be a good idea to add a comment that goes like "this is a placeholder for demonstration purposes" near that proc.

If you're talking about why `rand` needs to be copied, then I think it's so that each thread's sequence of random numbers doesn't overlap with another thread's sequence.

Honestly, as someone who barely delves into multi-threading, documenting `skipRandomNumbers` was difficult for me so I had to rely on this comment which was really helpful: https://github.com/nim-lang/Nim/pull/10546#issuecomment-461541312

Hope that helps.

Reply:
but i don't understand the logic.
`var r = rand` => `r` isn't used after this assignment
`result += rand(1..10)` => this uses randState, not `r`

so I'm assuming you meant:
```nim
result += rand(r, 0..10)
```

Reply:
lol yeah oops